modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+      }
+    }
+
+    function getNestedValue(obj, path) {
+      return path.split('.').reduce(function (prev, curr) {
+        return prev[curr];
+      }, obj || this);
+    }
+
+    $scope.deleteInList = function (path, index) {
+      let newKeyValue = getNestedValue($scope.data, path).filter((_val, num) => num != index);
+      if (!newKeyValue.length) newKeyValue.push('');
+      updateObjectNestedProp(
+        $scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.addInList = function (path, value) {
+      let newKeyValue = getNestedValue($scope.data, path);
+      newKeyValue.push(value);
+      updateObjectNestedProp($scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.trigCorsMethod = function (method) {
+      if (!$scope.data.cors.allowMethods.includes(method))
+        $scope.data.cors.allowMethods.push(method);
+      else
+        $scope.data.cors.allowMethods = $scope.data.cors.allowMethods.filter((x) => x != method);
+    }
+
+    function saveContents(contentToSave) {
+      if ($scope.file) {
+        console.log('Save of XSACCESS called...');
+        let xhr = new XMLHttpRequest();
+        xhr.open('PUT', '/services/v4/ide/workspaces' + $scope.file);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState === 4) {
+            console.log('XSACCESS file saved: ' + $scope.file);
+          }
+        };
+        xhr.send(contentToSave);
+        messageHub.post({ data: $scope.file }, 'editor.file.saved');
+        messageHub.post({ data: 'Requested file [' + $scope.file + '] saved.' }, 'status.message');
+      } else {
+        console.error('XSACCESS file parameter is not present in the request');
+      }
+    }
+
+    $scope.createSaveContent = function () {
+      let contentsJSON = JSON.parse(angular.toJson($scope.data));
+      if (contentsJSON.cors.enabled) {
+        contentsJSON.cors.allowOrigin = contentsJSON.cors.allowOrigin.filter((x) => x.trim());
+        contentsJSON.cors.allowHeaders = contentsJSON.cors.allowHeaders.filter((x) => x.trim());
+        contentsJSON.cors.exposeHeaders = contentsJSON.cors.exposeHeaders.filter((x) => x.trim());
+      } else {
+        contentsJSON.cors = { enabled: false };
+      }
+      if (!contentsJSON.headers.enabled) {
+        contentsJSON.headers = { enabled: false };
+      } else {
+        contentsJSON.headers.customHeaders = contentsJSON.headers.customHeaders.map((x) => {
+          return { name: x.name, value: x.thevalue + (x.thevalue == 'ALLOW-FROM' ? " " + x.url : '') };
+        }).filter((x) => x.name.trim());
+      }
+      contentsJSON.authorization = contentsJSON.authorization.filter((x) => x.trim());
+      contentsJSON.mime_mapping = contentsJSON.mime_mapping.filter((x) => x.extension.trim() && x.mimetype.trim());
+      contentsJSON.rewrite_rules = contentsJSON.rewrite_rules.filter((x) => x.source.trim() && x.target.trim());
+      return JSON.stringify(contentsJSON, null, 4);
```

Comment:
Is this JSON shown to the user somewhere? If it's not, why prettify?

Reply:
Oh, I saw it below

Reply:
Sure, we use it ;) I would even suggest Preview JSON as a standard option in all form-based editors.